### PR TITLE
Simplify dirAdd and Ledger fetching and construction

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

File: src/ripple/app/ledger/impl/InboundLedger.cpp [hunk 1]
```diff
@@ -128,33 +154,63 @@ InboundLedger::~InboundLedger ()
...
+    info.drops = sit.get64 ();
+    info.parentHash = sit.get256 ();
+    info.txHash = sit.get256 ();
+    info.accountHash = sit.get256 ();
+    info.parentCloseTime = NetClock::time_point{NetClock::duration{sit.get32()}};
+    info.closeTime = NetClock::time_point{NetClock::duration{sit.get32()}};
+    info.closeTimeResolution = NetClock::duration{sit.get8()};
+    info.closeFlags = sit.get8 ();
+
+    return std::move(info);
```
Comment (seelabs) [inline src/ripple/app/ledger/impl/InboundLedger.cpp]: Don't move here; it inhibits RVO.

Comment (nbougalis) [inline src/ripple/app/ledger/impl/InboundLedger.cpp]: Yeah, Vinnie noticed this as well. I'll fix.

File: src/ripple/ledger/impl/View.cpp [hunk 2]
```diff
@@ -687,107 +687,114 @@ dirNext (ApplyView& view,
     return true;
 }
 
-std::function<void (SLE::ref, bool)>
+std::function<void (SLE::ref)>
 describeOwnerDir(AccountID const& account)
 {
-    return [account](std::shared_ptr<SLE> const& sle, bool)
+    return [&account](std::shared_ptr<SLE> const& sle)
```
Comment (seelabs) [inline src/ripple/ledger/impl/View.cpp]: I haven't audited the calls of `describeOwnerDir`, but capturing by value here is much safer. It's not obvious that the lifetime of the `account` parameter is longer than the lifetime of the return value. And even if we do now, this seem fragile for a small optimization.

**Comment by nbougalis** [comment]
(closing this temporarily - more work)

**Pull request closed without merging**